streams/src/test/java/org/apache/kafka/streams/kstream/KStreamBuilderTest.java

```diff
@@ -77,23 +77,41 @@ public void testNewName() {
         assertEquals("Y-0000000001", builder.newName("Y-"));
         assertEquals("Z-0000000002", builder.newName("Z-"));
 
-        KStreamBuilder newBuilder = new KStreamBuilder();
+        final KStreamBuilder newBuilder = new KStreamBuilder();
 
         assertEquals("X-0000000000", newBuilder.newName("X-"));
         assertEquals("Y-0000000001", newBuilder.newName("Y-"));
         assertEquals("Z-0000000002", newBuilder.newName("Z-"));
     }
 
+
+    @Test
+    public void shouldHaveSinkTopicForStreamTo() {
+        final KStream<String, String> source = builder.stream("topic-source");
+        source.to("topic-sink");
+
+        final MockProcessorSupplier<String, String> processorSupplier = new MockProcessorSupplier<>();
+
+        source.process(processorSupplier);
+
+        driver = new KStreamTestDriver(builder);
+        driver.setTime(0L);
+
+        driver.process("topic-source", "A", "aa");
```

Comment:
`KStreamTestDriver` fails if you use `.to()` -- thus, the test uses a simple `builder.stream(...).to(...)` pattern to verify that it does not crash anymore.

The underlying problem is, that a sink does call `KStreamTestDriver.process()` again -- we need this logic for internal repartitioning topics. However, for `.to()` there is no downstream processor and thus we get NPE.

If we allow to add the "sink" node a new source, it avoids the NPE, and because a sink does not have any children, it terminates.

But thinking about this a second time, it might be a better fix, to change `KStreamTestDriver $MockRecordCollector.send()` -- check the topic name, and only call `process()` for internal repartitioning topics.

\cc @wimvanleuven